### PR TITLE
Added support for long suffix domain

### PR DESCRIPTION
**Pull request by tinogomes**

### CODE REVIEW

File: lib/validators/constants.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 module Validators
-  EMAIL_FORMAT = /\A[a-z0-9]+([-._][a-z0-9]+)*(\+[^@]+)?@[a-z0-9]+([.-][a-z0-9]+)*\.[a-z]{2,4}\z/i
+  # Support long suffix domain, like: .museum, .fitness
+  EMAIL_FORMAT = /\A[a-z0-9]+([-._][a-z0-9]+)*(\+[^@]+)?@[a-z0-9]+([.-][a-z0-9]+)*\.[a-z]{2,}\z/i
```
Comment (fnando) [inline lib/validators/constants.rb]: My idea with this is being more strict about the rules (mainly accepting .com and other common TLDs); you can use `tld: true` for covering cases like this. Convince me that this is a good idea. :)

File: test/validators/validates_email_format_of_test.rb [hunk 2]
```diff
@@ -19,10 +19,10 @@ class ValidatesEmailFormatOfTest < Minitest::Test
 
   INVALID_EMAILS.each do |email|
     test "rejects #{email.inspect} as a valid email" do
-      user = User.new(:email => "invalid", :corporate_email => "invalid")
+      user = User.new(:email => email, :corporate_email => email)
```
Comment (fnando) [inline test/validators/validates_email_format_of_test.rb]: Excellent catch. This should definitely get merged!

File: test/validators/validates_email_format_of_test.rb [hunk 3]
```diff
@@ -19,10 +19,10 @@ class ValidatesEmailFormatOfTest < Minitest::Test
...
       refute user.valid?
 
-      user = Person.new(:email => "invalid")
+      user = Person.new(:email => email)
```
Comment (fnando) [inline test/validators/validates_email_format_of_test.rb]: Same!

File: lib/validators/constants.rb (hunk 1 above)
Comment (tinogomes) [inline lib/validators/constants.rb]: And `tld: true` works only for hostname validator.

I agree about to use strict rules, but IMO, long suffix domains is required about this rules today. So, to be strict about long suffix domains, or we change this basic rule `/...\.[a-z]{2,4}\z/i` to `...\.(?:#{tld_long_suffix_domains_list.join("|")})\z/i`, or apply the `tld` options for email validator. Even for this last one, we need to be less strict on domain suffix validation.

**Comment by tinogomes** [comment]
Replaced by PR #8

**Pull request closed without merging**